src/main/java/com/training/tinyurl/util/TinyUrlGenerator.java

```diff
@@ -0,0 +1,16 @@
+package com.training.tinyurl.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.Base64;
+
+public class TinyUrlGenerator {
+    private TinyUrlGenerator(){}
+    public static String createShortCode(String longUrl) throws NoSuchAlgorithmException {
+        String salt = String.valueOf(Instant.now().getEpochSecond() + longUrl);
```

Comment:
According to design doc, time will only get appended in case of collision. Collision can still happen, so you need to retry atleast thrice. If still collision happens, log and return error 500.